src/Surfnet/ServiceProviderDashboard/Domain/Entity/Service.php

```diff
@@ -147,7 +147,7 @@ class Service
     /**
      * @var PrivacyQuestions
      *
-     * @ORM\OneToOne(targetEntity="PrivacyQuestions", mappedBy="service")
+     * @ORM\OneToOne(targetEntity="PrivacyQuestions", mappedBy="service", cascade={"remove"}, orphanRemoval=true)
```

Comment:
I don't like this solution because it goes against the application architecture as defined in the documentation of the project. Also this behavior is entirely different then the removal of other related entities (see the removal of an `Entity`) in a `Service`. So I would suggest to remove this responsibility away from the entity and remove it to the command handler.

Reply:
Yes, this would make the application better. For example, we would then also support table engines that do not support foreign key relations. On the other hand, during design time we accepted the cluttering of the domain layer with some specific Doctrine (infra) annotations. Because it worked really well for us. Alternatively, we could have used YML config for doctrine entity configuration.

My suggestion is to put this in a technical debt ticket. For now, I don't see good reason to change this. And when we would, we should fix it throughout the lines.

Reply:
Added this thread to the 'remove drafts' story:
https://www.pivotaltracker.com/story/show/166004131/comments/203097098